### PR TITLE
[postponed] Drop support for EOL Python (2.6 and) 3.3

### LINKED ISSUE #1016: Drop support for EOL Python 2.6 and 3.3?
It looks like significant effort is going into maintaining support for EOL Python versions. When they reach EOL that means they're no longer receiving security updates.

image

https://en.wikipedia.org/wiki/CPython#Version_history

Here's the pip installs for jedi from PyPI for the last month (via `pypinfo --percent --pip --markdown jedi pyversion`).

| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 3.6            |   44.4% |        155,231 |
| 3.5            |   27.5% |         96,124 |
| 3.4            |   22.2% |         77,788 |
| 2.7            |    5.2% |         18,142 |
| 3.7            |    0.4% |          1,559 |
| 3.3            |    0.2% |            800 |
| 2.6            |    0.0% |             30 |
| 3.2            |    0.0% |              5 |

From these numbers, it looks like 2.6 and 3.3 can be safely dropped.

This would allow the code to be modernised and ease the maintenance burden quite a bit.

Thoughts?

### PR DESCRIPTION
**Pull request by hugovk**
Edit: rebased on master, so there's just the one commit needed here.

### CODE REVIEW

**Comment by hugovk** [comment]
Travis CI failure is a temporary network issue, it passed on my fork: https://travis-ci.org/hugovk/jedi/builds/325357381.

**Comment by hugovk** [comment]
Rebased on master.

File: tox.ini [hunk 1]
```diff
@@ -30,30 +29,20 @@ commands =
...
-    numpydoc
     {[testenv]deps}
 [testenv:py36]
 deps =
-    numpydoc
     {[testenv]deps}
 [testenv:cov]
 deps =
     coverage
-    numpydoc
```
Comment (blueyed) [inline tox.ini]: Why is numpydoc removed here (and above)?

Comment (hugovk) [reply tox.ini]: Oops, that'll be a mistake from rebasing. I'll revert it!

Comment (davidhalter) [reply tox.ini]: Thanks for finding this one!

**Comment by davidhalter** [comment]
Thanks for the rebase. I merged it finally. This means that the next version will not support Python 3.3.

**Pull request merged**